CIPs/0008.md

```diff
@@ -28,32 +28,35 @@ We propose to add another claim, called the `STORAGE` claim:
 {
   "type": "STORAGE",
   "timestamp": 1588886114907,
-  "address": "https://example.com/celo_storage"
+  "address": "https://example.com/celo_storage",
+  "filterDataAddresses": ".*"
 }
 ```
 
-Alternatively, this could be set directly on the `Accounts` smart contract directly, causing off-chain storage to effectively replace the existing metadata functionality. This would have the benefit of not duplicating similar functionality.
+This http address should be referred to as the root. The expectation here is that data can accessed via `root/${dataAddress}`. A root can specify a REGEX with the `filterDataAddresses` key to indicate what kind of data it stores. An account can make multiple `STORAGE` claims, both to increase availability, as well as effectively sharding this off-chain data. This CIP does not specify what to do in case of conflicts.
 
-Either way, this http address should be referred to as the root. The expectation here is that data can accessed via `root/${dataAddress}`. For private data, this should be encrypted to the recipient. Here are some common ways of modeling applications into this format:
+Here are some common ways of modeling applications into this format:
 
 - For transaction metadata, the account can encrypt the metadata and put it in a file with the transaction hash as the metadata. Further CIPs could specify a standard for thee structure of that metadata.
-    - `dataAddress`: `transactionHash`
+    - `dataAddress`: `/transactions/transactionHash`
     - file contents: Encrypted, structured transaction metadata (comments, or otherwise)
 - Data in relation to other accounts could be communicated to other accounts with:
-    - `dataAddress`: `otherAccountAddress`
+    - `dataAddress`: `/accounts/otherAccountAddress`
     - file contents: Structured data such as:
         - `sessionKeys`: The session key(s) encrypted with dataEncryptionKey of `otherAccount` on `Accounts.sol`
         - `attestations`: These could be signed data attestating things such an audit or badges. An attested account could also store the attestations from the attestor here, as a client can just verify the signature of the attestor.
 - Profile information could be stored on this system with:
-    - `dataAddress`: `profileName` or `profilePicture`
+    - `dataAddress`: `/profile/name` or `/profile/picture`
     - file contents: the data unencrypted, or encrypted to session keys available to the recipients
 - Releases:
-    - `dataAddress`: `hashOfRelease`
+    - `dataAddress`: `/releases/hashOfRelease`
     - file contents: The signature of the key on the hash of the release
 
+Application developers should consider using standards such as JSON-LD to not reinvent the wheel
```

Comment:
I've taken it out for now. I was originally thinking that these paths could represent semantic links indeed, but the "host" would have to be some synthetic host like `celo://offchain-${accountAddress}/`. However since that is not core to this CIP I'm gonna leave it as a future improvement

I do think it would be valuable as it aids in making the data more interoperable